### PR TITLE
Load mash widgets onto the page

### PR DESCRIPTION
**Pull request by mdedys**
## Description

- Dynamically inject scripts for widgets when SDK is created
- OnInit if autoHide is enabled, only show the Mash Wallet if there is a Mash Element on the page.

### Linked GitHub Tickets

- Closes [Automatically load web component scripts through MashSDK](https://github.com/getmash/mash/issues/1424)
- Closes [Load Mash Wallet when there is Mash Elements on the page](https://github.com/getmash/mash/issues/1425)

### CODE REVIEW

File: packages/client-sdk/src/widgets/widgets.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+/**
+ * Inject widgets into the page
+ * @param baseURL Base URL of endpoint where widgets are hosted
+ */
+export function injectWidgets(baseURL: string) {
+  const scripts = Object.values(Widgets).map(value => {
+    const script = window.document.createElement("script");
+    script.async = true;
```
Comment (mdedys) [inline packages/client-sdk/src/widgets/widgets.ts:38]: @jstol I am always so torn here. In the case of web components, does async make sense? We don't really care the order or DOM to be ready for the `customDefinition` to be loaded?

Comment (mdedys) [reply packages/client-sdk/src/widgets/widgets.ts:38]: In the future we could also support, just selecting which widgets to bring in. No need to bring in everything if we are only using the "VideoPlayer" for example. I think this approach makes that easier.

Comment (jstol) [reply packages/client-sdk/src/widgets/widgets.ts:38]: Was just wondering if the headaches of giving that flexibility was worth it, but I'm cool leaving it as-is if we think that makes the most sense given how it's currently set up

Comment (jstol) [reply packages/client-sdk/src/widgets/widgets.ts:38]: Mostly the potential perf impact of needing to loop through and inject all of the `script` tags & have the browser download them individually. But maybe it's negligible?

Comment (mdedys) [reply packages/client-sdk/src/widgets/widgets.ts:38]: I could see it being a problem if we have 50 components but for the time being it is negligible. That’s how regular web components are already loaded across the web. As we delve deeper into the floating widget world we will probably tweak this with a more constrained list of components. I don’t think the end goal will be to always include everything.

Comment (jstol) [reply packages/client-sdk/src/widgets/widgets.ts:38]: Want the defs to get registered as quickly as possible without blocking parsing to download so feels like the right choice

File: packages/client-sdk/src/Mash.ts [hunk 2]
```diff
@@ -50,6 +60,13 @@ class Mash {
   }
 
   init(settings: MashSettings) {
+    if (this.config.autoHide && !isWidgetOnPage()) {
+      console.info(
+        "MASH] No mash elements found on page. Mash Wallet is hidden",
```
Comment (cartonioh) [inline packages/client-sdk/src/Mash.ts:65]:
```suggestion
        "[MASH] No mash elements found on page. Mash Wallet is hidden",
```

File: packages/client-sdk/src/widgets/preconnect.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Preconnect to given URL. https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/preconnect
```
Comment (cartonioh) [inline packages/client-sdk/src/widgets/preconnect.ts:2]: TIL thanks for the link

**Review by cartonioh — APPROVED** [review]

File: packages/client-sdk/src/widgets/preconnect.ts (hunk 3 above)
Comment (jstol) [reply packages/client-sdk/src/widgets/preconnect.ts:2]: Same ty

**Review by jstol — APPROVED** [review]

**Pull request merged**